src/Features/CSharp/Portable/ForeachToFor/CSharpForeachToForCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,160 @@
+                }
+
+                if (trivia.Kind() != SyntaxKind.WhitespaceTrivia &&
+                    trivia.Kind() != SyntaxKind.EndOfLineTrivia)
+                {
+                    // we don't know what to do with these comments
+                    return null;
+                }
+            }
+
+            return foreachStatement;
+        }
+
+        protected override (SyntaxNode start, SyntaxNode end) GetForeachBody(SyntaxNode node)
+        {
+            var foreachStatement = (ForEachStatementSyntax)node;
+            return (foreachStatement.Statement, foreachStatement.Statement);
+        }
+
+        protected override void ConvertToForStatement(SemanticModel model, ForeachInfo foreachInfo, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var parseOption = model.SyntaxTree.Options;
+            var foreachStatement = (ForEachStatementSyntax)foreachInfo.ForeachStatement;
+
+            // first, see whether we need to introduce new statement to capture collection
+            var foreachCollectionExpression = GetForeachCollection(foreachStatement);
+            var collectionVariableName = foreachCollectionExpression.ToString();
+            if (foreachInfo.RequireCollectionStatement)
+            {
+                collectionVariableName = CreateUniqueMethodName(model, foreachStatement, "list");
+
+                var collectionStatement = SyntaxFactory.LocalDeclarationStatement(
+                    SyntaxFactory.VariableDeclaration(
+                        SyntaxFactory.IdentifierName("var"),
+                        SyntaxFactory.SingletonSeparatedList(
+                            SyntaxFactory.VariableDeclarator(
+                                SyntaxFactory.Identifier(collectionVariableName).WithAdditionalAnnotations(RenameAnnotation.Create()),
+                                default,
+                                SyntaxFactory.EqualsValueClause(
+                                    foreachInfo.RequireExplicitCast ?
+                                        SyntaxFactory.CastExpression(
+                                            SyntaxFactory.ParseTypeName(foreachInfo.ExplicitCastInterface).WithAdditionalAnnotations(Simplifier.Annotation),
+                                            SyntaxFactory.ParenthesizedExpression(foreachCollectionExpression)) : 
+                                        foreachCollectionExpression)))));
+
+                collectionStatement =
+                    collectionStatement.ReplaceToken(collectionStatement.GetFirstToken(), collectionStatement.GetFirstToken().WithLeadingTrivia(foreachStatement.ForEachKeyword.LeadingTrivia));
+
+                editor.InsertBefore(foreachStatement, collectionStatement);
+            }
+
+            // create new index varialbe name
+            var indexString = CreateUniqueMethodName(model, foreachStatement.Statement, "i");
```

Comment:
what if "i" already exists? it's not uncommon that this would happen in a nested loop

Reply:
i is just base. it will add  number to make it unique.

Reply:
oh i see, i don't know how i overlooked that, sorry (but CreateUniqueMethodName might be a little misleading because it's not just for methods)

Reply:
agree. I should change name of the method.